Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterProcess.cs

```diff
@@ -43,13 +44,21 @@ internal sealed class DebugAdapterProcess {
             _webBrowserUrl = webBrowserUrl;
             _targetHostProcess = _targetInterop.ExecuteCommandAsync(pythonExePath, "\"" + _debuggerAdapterDirectory + "\"");
 
-            if (!string.IsNullOrEmpty(webBrowserUrl) && Uri.TryCreate(webBrowserUrl, UriKind.RelativeOrAbsolute, out var uri)) {
-                OnPortOpenedHandler.CreateHandler(uri.Port, null, null, () => _targetHostProcess.HasExited, LaunchBrowserDebugger);
+            if (!string.IsNullOrEmpty(webBrowserUrl) && Uri.TryCreate(webBrowserUrl, UriKind.RelativeOrAbsolute, out _)) {
+                Timer timer = new Timer(5000);
```

Comment:
Timer needs to be `Dispose()`. I suppose you could do that in the elapsed handler.

Not sure why you decided to go with this Timer class rather than the one in System.Threading that we use everywhere... Regardless all Timer classes require disposing.

Reply:
Changed